### PR TITLE
Add Full Enemy Control

### PR DESCRIPTION
**Pull request by xAstroBoy**
As Per requested, in a different PR.

### CODE REVIEW

**Comment by xAstroBoy** [comment]
@winstxnhdw everything comes from ControlCompany and has been deeply rewrotten to be used in PossessionMod, which i tested each and one of these enemies, and it 100% works.

File: lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                if (instance.CanGrabScrap(item)) {
+                    if (item.TryGetComponent(out NetworkObject network)) {
+                        float dist = Vector3.Distance(item.transform.position, instance.transform.position);
+                        if (dist < float.PositiveInfinity) {
+                            return network;
+                        }
+                    }
+                }
+            }
+        }
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs:31]: Fantastic work. Could you reduce the nesting here and other similar places? I believe I saw some names that were too generic as well.

**Review by winstxnhdw — CHANGES_REQUESTED** [review]

File: lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs (hunk 1 above)
Comment (xAstroBoy) [reply lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs:31]: i have been on this for 5 hours, i tried to reduce the nesting as much i could, i dont think is possible unless you want to give a try

**Comment by winstxnhdw** [comment]
Looks like it won't be too hard to merge this one.

File: lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs (hunk 1 above)
Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs:31]: Take a break. Also, you can use the `continue` statement.

Comment (xAstroBoy) [reply lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs:31]: > Take a break. Also, you can use the `continue` statement.

why i need to use the continue statement when i return the object from that method? im confused lol

Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs:31]: ``cs
if (!Search[i].TryGetComponent(out GrabbableObject item)) continue;
 ```

This is known as an if-guard.

**Pull request closed without merging**